rosbag_snapshot/src/snapshotter.cpp

```diff
@@ -195,14 +179,9 @@ bool MessageQueue::preparePush(int32_t size, ros::Time const& time, const std::s
     while (queue_.size() != 0 && size_ + size > options_.memory_limit_)
       _pop();
 
-  // If topic is latched, ignore duration limit but remove all previous messages from the specified callerid
-  if (isLatched())
-  {
-    removeCallerid(callerid);
-  }
   // If duration limit is enforced, remove elements from front of queue until duration limit would be met once message
   // is added
-  else if (options_.duration_limit_ > SnapshotterTopicOptions::NO_DURATION_LIMIT && queue_.size() != 0)
+  if (options_.duration_limit_ > SnapshotterTopicOptions::NO_DURATION_LIMIT && queue_.size() != 0)
```

Comment:
Is there no `empty()` method for `queue_`?

Reply:
There is, but that came from [upstream rosbag_snapshot](https://github.com/ros/rosbag_snapshot/blob/6809260e1205004c74d409e52ae7d22954bdb269/rosbag_snapshot/src/snapshotter.cpp#L159) and in fact, this just changes it back to an `if` like it is in upstream instead of `else if`. I've been trying to limit my changes to only what is required if that makes sense.

Reply:
It does makes sense, thanks.